flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedSequenceNumberingViewReader.java

```diff
@@ -100,8 +106,19 @@ public void addCredit(int creditDeltas) {
         numCreditsAvailable += creditDeltas;
     }
 
+    @Override
+    public boolean needAnnounceBacklog() {
+        return initialCredit == 0 && numCreditsAvailable == 0;
+    }
+
     @Override
     public void resumeConsumption() {
+        if (initialCredit == 0) {
+            // reset available credit if no exclusive buffer is available at the
+            // consumer side for all floating buffers must have been released
+            numCreditsAvailable = 0;
```

Comment:
Do we do it because we know that all floating buffers would be released before the checkpoint when `initialCredit == 0`?

Reply:
Yes, you are right.